h5pxblock/h5pxblock.py

```diff
@@ -280,17 +280,21 @@ def result_handler(self, data, suffix=''):
         except BaseException as exp:
             log.error("Error while marking completion %s", exp)
 
-        try:
-            if self.has_score and data['result'] and data['result']['score']:
-                grade_dict = {
-                    'value': data['result']['score']['raw'],
-                    'max_value': data['result']['score']['max'],
-                    'only_if_higher': True,
-                }
+        if self.has_score and data['result'] and data['result']['score']:
+            grade_dict = {
+                'value': data['result']['score']['raw'],
+                'max_value': data['result']['score']['max'],
+                'only_if_higher': True,
+            }
+            try:
                 self.runtime.publish(self, 'grade', grade_dict)
                 save_score = True
-        except BaseException as exp:
-            log.error("Error while publishing score %s", exp)
+            except TypeError:
```

Comment:
Is there a way to avoid this error altogether? Can we know from the grade_dict when is the first time an answer is uploaded? Or is the error our best guess?

Reply:
That was one of the options I considered, however, the `grade_dict` can have any value in the `value` key, so it would not be useful to know if it is the first time an answer is uploaded. 

I am not sure if there is a way to get the score before publishing and maybe avoid the error. At the moment, I think this would be the best option.

Reply:
That sounds good to me!